GPSTest/src/main/java/com/android/gpstest/GpsTestActivity.java

```diff
@@ -980,10 +980,10 @@ private synchronized void gpsStart() {
             showProgressBar();
 
             // Show Toast only if the user has set minTime or minDistance to something other than default values
-            if (minTime != (long) (Double.valueOf(getString(R.string.pref_gps_min_time_default_sec))
+            if (minTime != (Long.parseLong(getString(R.string.pref_gps_min_time_default_sec))
```

Comment:
This changes the behavior. Let's leave this as parsing a `Double` and then converting to `long` (so using `Double.parseDouble()` is fine here). It's been a while since I wrote this but IIRC I did this so I could change the default values to decimal in the resources without needing to change the code here.

Reply:
Makes perfect sense.. And that's what code review is for! 8-)

I've reverted it back to the original version..